packages/rum-core/test/eventFormatValidation.ts

```diff
@@ -0,0 +1,54 @@
+import { registerCleanupTask } from '@datadog/browser-core/test'
+import type { TimeStamp, Context } from '@datadog/browser-core'
+import { combine } from '@datadog/browser-core'
+import type { LifeCycle, RawRumEventCollectedData } from '../src/domain/lifeCycle'
+import { LifeCycleEventType } from '../src/domain/lifeCycle'
+import type { RawRumEvent, RumContext } from '../src/rawRumEvent.types'
+import { validateRumFormat } from './formatValidation'
+
+export function collectAndValidateRawRumEvents(lifeCycle: LifeCycle) {
+  const rawRumEvents: Array<RawRumEventCollectedData<RawRumEvent>> = []
+  const subscription = lifeCycle.subscribe(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, (data) => {
+    rawRumEvents.push(data)
+    validateRumEventFormat(data.rawRumEvent)
+  })
+  registerCleanupTask(() => {
+    subscription.unsubscribe()
+  })
+
+  return rawRumEvents
+}
+
+function validateRumEventFormat(rawRumEvent: RawRumEvent) {
```

Comment:
**💬 suggestion:** ‏Even though `testSetupBuilder` will be gone later, avoid duplicating code and export/import the `validateRumEventFormat` function.